src/transformStateWithClones.js

```diff
@@ -64,8 +64,37 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const states = [];
+  let newState = { ...state };
+
+  for (const { operation, properties } of transforms) {
+    switch (operation) {
+      case 'addProperties':
+        for (const propAddKey in properties) {
+          newState[propAddKey] = properties[propAddKey];
+        }
+        states.push(newState);
+        newState = { ...newState };
```

Comment:
```suggestion
        states.push({ ...newState });
```
Не надо добавлять ссылку на обьект в массив, лучше добавить его копию.
        states.push({ ...newState }); лучше вынести в конец цикла for что б не дублировать.